### ISSUE TITLE
Parsing error on AsyncGeneratorFunction

### ISSUE THREAD

**Issue by aabfred** [eslint/eslint#10406]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 10.1.0
* **npm Version:** 6.0.1

**What parser (default, Babel-ESLint, etc.) are you using?** eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "es6": true,
        "node": true,
        "mocha": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 2017
    },
    "rules": {
        "indent": [
            "warn",
            4,
            { "SwitchCase": 1 }
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "warn",
            "double",
            { "allowTemplateLiterals": true, "avoidEscape": true }
        ],
        "semi": [
            "error",
            "always"
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function* f1(){}
async function f2(){}
async function* f3(){}
module.exports = [ f1, f2, f3 ];
```

```bash
node_modules/.bin/eslint ./bug.js
```

**What did you expect to happen?**
Empty report

**What actually happened? Please include the actual, raw output from ESLint.**
/ANY/bug.js
  3:15  error  Parsing error: Unexpected token *

✖ 1 problem (1 error, 0 warnings)

**Comment by Aladdin-ADD** [comment]
To allow async functions, you need config `ecmaVersion: 2018`
```js
    "parserOptions": {
        "ecmaVersion": 2018
    },
```

**Comment by aabfred** [comment]
Thank you
